clearwater_config_manager/sas_json_plugin.py

```diff
@@ -50,5 +75,53 @@ def on_config_changed(self, value, alarm):
             run_command(["/usr/share/clearwater/bin/reload_sas_json"])
             alarm.update_file(_file)
 
+    def resolve_domain_address(self, addr, dns_server=None): # pragma: no cover
+        ip_addrs = []
+
+        # Ensure we're using the correct dns server.
+        if dns_server:
+            dns.resolver.override_system_resolver()
+            dns.resolver.get_default_resolver().nameservers = dns_server.split(',')
+
+        try:
+            resolver = dns.resolver.get_default_resolver()
+
+            # timeout is the time spent waiting for each DNS server.
+            # lifetime is the time spent waiting for a response overall.
+            # By setting timeout to 2 and lifetime to 4, we allow time for us to
+            # check at least 2 of our DNS servers before we give up.
+            resolver.timeout = 2.0
+            resolver.lifetime = 4.0
+            ip_addrs = resolver.query(addr)
+        except Exception:
+            # If something goes wrong, we can't resolve the address so we'll
+            # just return an empty list.
+            pass
+
+        # Revert changes to the dns server
+        if dns_server:
+            dns.resolver.restore_system_resolver()
+            dns.resolver.reset_default_resolver()
+
+        return ip_addrs
+
+    def resolve_domain_address_in_namespace(self, addr, is_signaling): # pragma: no cover
+        dns_answers = []
+
+        if is_signaling == 'Y':
+            sig_ns = os.environ.get('signaling_namespace')
+            sig_dns = os.environ.get('signaling_dns_server')
+
+            if sig_ns:
+                with Namespace('/var/run/netns/' + sig_ns, 'net'):
```

Comment:
This won't work - you haven't done `from nsenter import Namespace` - I'd have expected `make full_test` in clearwater-etcd to have caught this.

Reply:
I suspect that line was never hit (as make full_test passed), I'm in the process of patching this and testing it on a live system now.